llvm/include/llvm/MCCAS/MCCASObjectV1.h

```diff
@@ -568,6 +571,9 @@ class MCCASReader {
   Expected<uint64_t> materializeGroup(cas::ObjectRef ID);
   Expected<uint64_t> materializeSection(cas::ObjectRef ID, raw_ostream *Stream);
   Expected<uint64_t> materializeAtom(cas::ObjectRef ID, raw_ostream *Stream);
+  Expected<uint64_t> reconstructSection(SmallVector<char, 0> &AtomBuffer,
```

Comment:
`SmallVectorImpl`

Reply:
This should also be changed to SectionBuffer actually, shouldn't it be a MutableArrayRef now that I think about it?

Reply:
It depends: if the intent is to grow, then `SmallVectorImpl`. If the intent is to mutate without growing, then just `MutableArrayRef`. If the intent is to read only, then `ArrayRef`. But never `SmallVector` 😅

Reply:
The intent is to grow, so `SmallVectorImpl` it is, I didn't know that, thanks!